javascript/javascript3/week2/rewrite-time.js

```diff
@@ -0,0 +1,44 @@
+/*
+Rewrite setTimeout and navigator.geolocation.getCurrentPosition to promises. 
+The getCurrentPosition function is probably going to throw an error, but that is fine. 
+As long as you can use it as a promise.
+
+Example of usage
+
+*/
+
+setTimeoutPromise(3000).then(() => {
+  console.log("Called after 3 seconds");
+});
+
+getCurrentLocation()
+  .then((position) => {
+    // called when the users position is found
+    console.log(position);
+  })
+  .catch((error) => {
+    // called if there was an error getting the users location
+    console.log(error);
+  });
+
+// console.log(`this is inside of the async function`);
+async function getCurrentLocation() {
+  return new Promise((resolve) => {
+    navigator.geolocation.getCurrentPosition(function (position) {
+      const latitue = position.coords.latitude;
+      const longitute = position.coords.longitude;
+      position = { latitue, longitute };
+      resolve(position);
+    });
+  });
+}
+
+async function setTimeoutPromise(sec) {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve();
+      console.log(`This should be shown in 3 sec`);
```

Comment:
there is a small mistake here. resolve(..) expects an argument. For example resolve('timeout triggered after x seconds') For more details, see the following error message:

Expected 1 arguments, but got 0. Did you forget to include 'void' in your type argument to 'Promise'?ts(2794)